front/app/containers/ProjectsShowPage/timeline/index.tsx

```diff
@@ -110,27 +115,22 @@ const ProjectTimelineContainer = memo<Props & WithRouterProps>(
     }, []);
 
     useEffect(() => {
-      if (
-        !isNilOrError(location) &&
-        !isNilOrError(phases) &&
-        phases.length > 0
-      ) {
+      setPhaseURL(selectedPhase, phases, project, locale);
+    }, [selectedPhase]);
+
+    useEffect(() => {
+      if (!isNilOrError(phases) && phases.length > 0) {
         const currentPhase = getCurrentPhase(phases);
         const firstPhase = getFirstPhase(phases);
         const lastPhase = getLastPhase(phases);
         const lastPastPhase = getLastPastPhase(phases);
 
-        // if, coming from the siteMap, a phase url parameter was passed in, we pick that phase as the default phase,
-        // then remove the param so that when the user navigates to other phases there is no mismatch
-        if (isString(location?.query?.phase)) {
-          const phase = phases.find(
-            (phase) => phase.id === location.query.phase
-          );
-
-          if (phase) {
-            window.history.replaceState(null, '', location.pathname);
-            selectPhase(phase);
-          }
+        // if a phase parameter was provided, and it is valid, we set that as phase.
+        // otherwise, use the most logical phase
+        if (isValidPhase(phaseParam, phases)) {
+          const phaseNumber = Number(phaseParam);
+          const phaseIndex = phaseNumber - 1;
+          selectPhase(phases[phaseIndex]);
         } else if (currentPhase) {
           selectPhase(currentPhase);
         } else if (
```

Comment:
It's a very small detail, but here you could potentially use `getLatestRelevantPhase` (found in the same file as `getCurrentPhase`) which internally will try to figure out the 'best' phase if there is no current phase (e.g. the last active phase or the last phase if the project has ended).